src/epm.rs

```diff
@@ -62,6 +66,54 @@ impl std::fmt::Display for EpmConstant {
 	}
 }
 
+/// Represent voters that may be trimmed
+///
+/// The trimming works by removing the voter with the least amount of stake
+///
+/// It's using an internal `BTreeMap` to determine which voter to remove next
+/// and the voters Vec can't be sorted because the EPM pallet will index into it
+/// when checking the solution.
+#[derive(Debug, Clone)]
+pub struct TrimmedVoters {
+	voters: Voters,
+	voters_by_stake: BTreeMap<VoteWeight, usize>,
+}
+
+impl TrimmedVoters {
+	/// Create a new `TrimmedVotes`.
+	pub fn new(voters: Voters) -> Self {
+		let mut voters_by_stake = BTreeMap::new();
+
+		for (idx, (_voter, stake, _supports)) in voters.iter().enumerate() {
+			voters_by_stake.insert(*stake, idx);
+		}
+
+		Self { voters, voters_by_stake }
+	}
+
+	/// Trim the next voter with the least amount of stake.
```

Comment:
Have we tried to trim more than one voter in each iteration? This may reduce the number of iterations required to finalise the trimming. Maybe we could have an heuristic where depending on how much the solution exceeds the limit, trim 1 or more voters in one go.

This probably should not block this PR, but perhaps addressed in a subsequent PR if helpful.

Reply:
We could also calculate the difference between the `weight_trimmed`/`length_trimmed` calculated in  `prepare_election_result_with_snapshot` and use that as an heuristic of how much trimming will be required. Then we can use that to request the trimming of N voters in one go, if there's still a long way to trim.

Reply:
That's good point, I think we should merge this PR and create a issue for your idea.

If there's much trimming to do then yeah this would be very slow as it does it one-by-one.